### ISSUE TITLE
Unexpected warning from `Kernel#printf` when width or precision is given by an asterisk `*`

### ISSUE THREAD

**Issue by nobu** [oracle/truffleruby#4046]
```console
$ truffleruby -v -e 'printf "[%*d]\n", 3, 42'
truffleruby 25.0.0, like ruby 3.3.7, Oracle GraalVM JVM [arm64-darwin20]
-e:1: warning: too many arguments for format string
[ 42]
$ truffleruby -v -e 'printf "[%.*d]\n", 3, 42'
truffleruby 25.0.0, like ruby 3.3.7, Oracle GraalVM JVM [arm64-darwin20]
-e:1: warning: too many arguments for format string
[042]
```

```console
$ ruby3.3 -v -e 'printf "[%*d]\n", 3, 42'
ruby 3.3.10 (2025-10-23 revision 343ea05002) [arm64-darwin25]
[ 42]
$ ruby3.3 -v -e 'printf "[%.*d]\n", 3, 42'
ruby 3.3.10 (2025-10-23 revision 343ea05002) [arm64-darwin25]
[042]
```

**Comment by eregon** [comment]
It's already fixed on master since https://github.com/truffleruby/truffleruby/pull/3944, it will be in the next release.
```
$ truffleruby -v -e 'printf "[%*d]\n", 3, 42'
truffleruby 33.0.0-dev-bb226b84 (2025-12-01), like ruby 3.3.7, Oracle GraalVM JVM [x86_64-linux]
[ 42]
truffleruby -v -e 'printf "[%*d]\n", 3, 42'  9.63s user 0.84s system 460% cpu 2.273 total
```

**Comment by eregon** [comment]
TruffleRuby 33 [has been released](https://truffleruby.dev/blog/truffleruby-33-is-released) with this fix.
